addons/mail/static/src/components/message/message.xml

```diff
@@ -3,46 +3,44 @@
 
     <t t-name="mail.Message" owl="1">
         <t t-if="messageView">
-            <div class="o_Message position-relative pt-1"
+            <div class="o_Message position-relative py-1"
                 t-att-class="{
                     'o_Message_active': messageView.isActive,
                     'o-clicked': messageView.messagingAsClickedMessageView,
                     'o-discussion': messageView.message.isDiscussionOrNotification,
                     'o-has-message-selection': messageView.messageListViewMessageViewItemOwner and messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.replyingToMessageView,
                     'o-highlighted': messageView.message.isHighlighted or messageView.isHighlighted,
+                    'o-isReplyHighlighted bg-white shadow-lg': messageView.isHighlighted,
                     'o-isDeviceSmall': messaging.device.isSmall,
                     'o-not-discussion': !messageView.message.isDiscussionOrNotification,
                     'o-notification': messageView.message.message_type === 'notification',
                     'o-selected': messageView.isSelected,
-                    'o-squashed': messageView.isSquashed,
+                    'o-squashed pt-1': messageView.isSquashed,
                     'o-starred': messageView.message.isStarred,
-                    'mt-3': !messageView.isSquashed and messageView.messageListViewMessageViewItemOwner,
-                    'bg-white': messageView.message.isDiscussionOrNotification and !messageView.isSelected and !(messageView.message.isHighlighted or messageView.isHighlighted) and !messageView.isActive,
-                    'bg-100': (!messageView.message.isDiscussionOrNotification and !messageView.isActive and !messageView.isSelected) or (messageView.isActive and !messageView.isSelected and messageView.message.isDiscussionOrNotification and !(messageView.message.isHighlighted or messageView.isHighlighted)),
-                    'bg-200': !messageView.message.isDiscussionOrNotification and messageView.isActive and !messageView.isSelected,
-                    'bg-400': messageView.isSelected and !(messageView.message.isHighlighted or messageView.isHighlighted),
+                    'o-currentAuthor': messageView.message.isCurrentUserOrGuestAuthor,
+                    'mt-1': !messageView.isSquashed,
```

Comment:
Can group `o-squashed pt-1` and `mt-1` together (same condition)

Reply:
`mt-1` has the opposite condition (`messageView.isSquashed` / `!messageView.isSquashed`).
We should not group them :)

Reply:
oops, nevermind 🤡